src/interactive/AccordionPanel.jsx

```diff
@@ -34,42 +32,41 @@ class AccordionPanel extends React.Component {
 	}
 
 	/**
+	 * 
+	 * @description calls the AccordionPanelGroups's callback to toggle open state
+	 * and render the `AccordionPanel` open or closed, sets height in state
+	 * @param {Event} e - the event object
 	 * @returns {undefined}
-	 *
-	 * Updates state to toggle `AccordionPanel` open and closed
 	 */
-	_handleToggle(){
-		if (this.props.setClickedPanel) {
-			this.props.setClickedPanel(this);
-		}
+	_handleToggle(e){
+		e.preventDefault();
+		e.stopPropagation();
 
+		const isToggledOpen = !this.props.isOpen;
+		this.props.setClickedPanel && this.props.setClickedPanel(this.props.clickId, isToggledOpen);
 		this.setState({
-			height: this.getHeight(!this.state.isOpen),
-			isOpen: !this.state.isOpen
+			height: this.getHeight(isToggledOpen)
```

Comment:
I think you can/should skip the `setState` here - it will be handled by the parent update of `isOpen` in `componentWillReceiveProps`.

In fact, do you need to keep track of `height` in `this.state` at all? Structurally, it would be nice if `height` _always_ corresponded to `this.props.isOpen` - it would never get out of sync, and the component wouldn't have to have _any_ internal `state`, and wouldn't have to implement `componentDidMount` or `componentWillReceiveProps`

Reply:
so I was thinking the same, however - the `contentEl` that is required to calculate height is a `ref` that it set after a height needs to be calculated.

setting the height in `state` in `componentDidMount` allows for the `ref` to be there and also since its in `state` triggers a necessary re-render.

I tried just using `props`, but if a panel is open by default, the ref isnt set in time with render, and the height will get calculated wrong, and since no state is getting updated in `componentDidMount` there is no re-render with the correct height.

thoughts?

Reply:
I'm probably not seeing all the moving pieces, but I think you can still avoid state, but use `componentDidMount` to call [`this.forceUpdate()`](https://reactjs.org/docs/react-component.html#forceupdate) - I think that will allow the `ref` to be set on initial render, and allow `height` to be calculated correctly the second time around.